src/ImageSharp/Metadata/Profiles/IPTC/IptcProfile.cs

```diff
@@ -264,34 +268,34 @@ private void Initialize()
 
             this.values = new Collection<IptcValue>();
 
-            if (this.Data == null || this.Data[0] != 0x1c)
+            if (this.Data == null || this.Data[0] != IptcTagMarkerByte)
             {
                 return;
             }
 
-            int i = 0;
-            while (i + 4 < this.Data.Length)
+            int offset = 0;
+            while (offset + 4 < this.Data.Length)
             {
-                if (this.Data[i++] != 28)
+                bool isValidTagMarker = this.Data[offset++] == IptcTagMarkerByte;
+                byte recordType = this.Data[offset++];
```

Comment:
I see `recordType` is unused but I'm struggling to correlate this with the spec. Got any pointers.

Reply:
section 1.5 iii: 
```
Octet 2 is the binary representation of the record number. Note that the
envelope record number is always 1, and that the application records are
numbered 2 through 6, the pre-object descriptor record is 7, the object record
is 8, and the post-object descriptor record is 9.
```

I was not sure what to do with this info. Maybe it would be wise to consider the value invalid if its not in the range of 1 to 9.

Reply:
> Maybe it would be wise to consider the value invalid if its not in the range of 1 to 9.

Probably wise.